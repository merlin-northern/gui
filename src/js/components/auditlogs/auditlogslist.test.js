import React from 'react';
import { prettyDOM } from '@testing-library/dom';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import AuditLogsList from './auditlogslist';
import { defaultState, undefineds } from '../../../../tests/mockData';

const mockStore = configureStore([thunk]);

describe('Auditlogs Component', () => {
  let store;
  beforeEach(() => {
    store = mockStore({ ...defaultState });
  });

  it('renders correctly', async () => {
    const { baseElement } = render(
      <MemoryRouter>
        <Provider store={store}>
          <AuditLogsList
            count={defaultState.organization.eventsTotal}
            items={defaultState.organization.events}
            loading={false}
            page={1}
            onChangeRowsPerPage={jest.fn}
            onChangePage={jest.fn}
            onChangeSorting={jest.fn}
            perPage={20}
            sortDirection="desc"
          />
        </Provider>
      </MemoryRouter>
    );

    const view = prettyDOM(baseElement.firstChild, 100000, { highlight: false })
      .replace(/id="mui-[0-9]*"/g, '')
      .replace(/aria-labelledby="(mui-[0-9]* *)*"/g, '')
      .replace(/\\/g, '');
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });
});
