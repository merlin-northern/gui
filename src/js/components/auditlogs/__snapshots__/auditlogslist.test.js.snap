// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auditlogs Component renders correctly 1`] = `
"<div>
  <div
    class=\\"fadeIn deploy-table-contain auditlogs-list\\"
  >
    <div
      class=\\"auditlogs-list-item auditlogs-list-item-header muted\\"
    >
      <div
        class=\\"columnHeader\\"
        style=\\"cursor: initial;\\"
      >
        User
      </div>
      <div
        class=\\"columnHeader\\"
        style=\\"cursor: initial;\\"
      >
        Action
      </div>
      <div
        class=\\"columnHeader\\"
        style=\\"cursor: initial;\\"
      >
        Type
      </div>
      <div
        class=\\"columnHeader\\"
        style=\\"cursor: initial;\\"
      >
        Changed
      </div>
      <div
        class=\\"columnHeader\\"
        style=\\"cursor: initial;\\"
      >
        More details
      </div>
      <div
        class=\\"columnHeader\\"
      >
        Time
        <svg
          aria-hidden=\\"true\\"
          class=\\"MuiSvgIcon-root sortIcon selected true\\"
          focusable=\\"false\\"
          viewBox=\\"0 0 24 24\\"
        >
          <path
            d=\\"M3 18h6v-2H3v2zM3 6v2h18V6H3zm0 7h12v-2H3v2z\\"
          />
        </svg>
      </div>
      <div />
    </div>
    <div
      class=\\"auditlogs-list\\"
    >
      <div
        class=\\"auditlogs-list-item clickable\\"
      >
        <div>
          string@example.com
        </div>
        <div
          class=\\"uppercased\\"
        >
          create
        </div>
        <div
          class=\\"capitalized\\"
        >
          user
        </div>
        <div>
          user@acme.com
        </div>
        <div
          class=\\"capitalized\\"
          style=\\"align-items: flex-start; white-space: pre-line;\\"
        >
          change1
        </div>
        <time
          datetime=\\"2019-01-01T12:10:22+00:00\\"
        >
          2019-01-01 12:10
        </time>
        <div
          class=\\"uppercased link-color bold\\"
        >
          view details 
          <svg
            aria-hidden=\\"true\\"
            class=\\"MuiSvgIcon-root\\"
            focusable=\\"false\\"
            viewBox=\\"0 0 24 24\\"
          >
            <path
              d=\\"M16.01 11H4v2h12.01v3L20 12l-3.99-4z\\"
            />
          </svg>
        </div>
      </div>
      <div
        class=\\"auditlogs-list-item clickable\\"
      >
        <div>
          string
        </div>
        <div
          class=\\"uppercased\\"
        >
          create
        </div>
        <div
          class=\\"capitalized\\"
        >
          deployment
        </div>
        <div>
          to production
        </div>
        <a
          href=\\"/deployments/finished?open=true&id=string\\"
        >
          View deployment
        </a>
        <time
          datetime=\\"2019-01-01T12:16:22+00:00\\"
        >
          2019-01-01 12:16
        </time>
        <div
          class=\\"uppercased link-color bold\\"
        >
          view details 
          <svg
            aria-hidden=\\"true\\"
            class=\\"MuiSvgIcon-root\\"
            focusable=\\"false\\"
            viewBox=\\"0 0 24 24\\"
          >
            <path
              d=\\"M16.01 11H4v2h12.01v3L20 12l-3.99-4z\\"
            />
          </svg>
        </div>
      </div>
      <div
        class=\\"auditlogs-list-item clickable\\"
      >
        <div>
          string@example.com
        </div>
        <div
          class=\\"uppercased\\"
        >
          open_terminal
        </div>
        <div
          class=\\"capitalized\\"
        >
          device
        </div>
        <div>
          a1
        </div>
        <a>
          View session log
        </a>
        <time
          datetime=\\"2019-01-01T12:10:22+00:00\\"
        >
          2019-01-01 12:10
        </time>
        <div
          class=\\"uppercased link-color bold\\"
        >
          view details 
          <svg
            aria-hidden=\\"true\\"
            class=\\"MuiSvgIcon-root\\"
            focusable=\\"false\\"
            viewBox=\\"0 0 24 24\\"
          >
            <path
              d=\\"M16.01 11H4v2h12.01v3L20 12l-3.99-4z\\"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class=\\"loaderContainer shrunk\\"
    >
      <div
        class=\\"loader\\"
      >
        <span
          class=\\"dot dot_1\\"
        />
        <span
          class=\\"dot dot_2\\"
        />
        <span
          class=\\"dot dot_3\\"
        />
        <span
          class=\\"dot dot_4\\"
        />
      </div>
    </div>
    <div
      class=\\"MuiTablePagination-root flexbox margin-top \\"
    >
      <div
        class=\\"MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters\\"
      >
        <div
          class=\\"MuiTablePagination-spacer flexbox no-basis\\"
        />
        <p
          class=\\"MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit\\"
          
        >
          Rows
        </p>
        <div
          class=\\"MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot\\"
        >
          <div
            aria-haspopup=\\"listbox\\"
            
            class=\\"MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input\\"
            
            role=\\"button\\"
            tabindex=\\"0\\"
          >
            20
          </div>
          <input
            aria-hidden=\\"true\\"
            class=\\"MuiSelect-nativeInput\\"
            tabindex=\\"-1\\"
            value=\\"20\\"
          />
          <svg
            aria-hidden=\\"true\\"
            class=\\"MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon\\"
            focusable=\\"false\\"
            viewBox=\\"0 0 24 24\\"
          >
            <path
              d=\\"M7 10l5 5 5-5z\\"
            />
          </svg>
        </div>
        <p
          class=\\"MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit\\"
        />
        <div
          class=\\"flexbox\\"
        >
          <button
            class=\\"MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled\\"
            disabled=\\"\\"
            tabindex=\\"-1\\"
            type=\\"button\\"
          >
            <span
              class=\\"MuiIconButton-label\\"
            >
              <svg
                aria-hidden=\\"true\\"
                class=\\"MuiSvgIcon-root\\"
                focusable=\\"false\\"
                viewBox=\\"0 0 24 24\\"
              >
                <path
                  d=\\"M18.41 16.59L13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z\\"
                />
              </svg>
            </span>
          </button>
          <button
            class=\\"MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled\\"
            disabled=\\"\\"
            tabindex=\\"-1\\"
            type=\\"button\\"
          >
            <span
              class=\\"MuiIconButton-label\\"
            >
              <svg
                aria-hidden=\\"true\\"
                class=\\"MuiSvgIcon-root\\"
                focusable=\\"false\\"
                viewBox=\\"0 0 24 24\\"
              >
                <path
                  d=\\"M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z\\"
                />
              </svg>
            </span>
          </button>
          <div
            class=\\"flexbox\\"
            style=\\"align-items: baseline;\\"
          >
            <div
              class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-marginDense\\"
              style=\\"min-width: 30px; max-width: 30px; margin-right: 10px;\\"
            >
              <div
                class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense\\"
              >
                <input
                  aria-invalid=\\"false\\"
                  class=\\"MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense\\"
                  type=\\"text\\"
                  value=\\"1\\"
                />
              </div>
            </div>
            / 1
          </div>
          <button
            class=\\"MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled\\"
            disabled=\\"\\"
            tabindex=\\"-1\\"
            type=\\"button\\"
          >
            <span
              class=\\"MuiIconButton-label\\"
            >
              <svg
                aria-hidden=\\"true\\"
                class=\\"MuiSvgIcon-root\\"
                focusable=\\"false\\"
                viewBox=\\"0 0 24 24\\"
              >
                <path
                  d=\\"M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z\\"
                />
              </svg>
            </span>
          </button>
          <button
            class=\\"MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled\\"
            disabled=\\"\\"
            tabindex=\\"-1\\"
            type=\\"button\\"
          >
            <span
              class=\\"MuiIconButton-label\\"
            >
              <svg
                aria-hidden=\\"true\\"
                class=\\"MuiSvgIcon-root\\"
                focusable=\\"false\\"
                viewBox=\\"0 0 24 24\\"
              >
                <path
                  d=\\"M5.59 7.41L10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z\\"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
